### PR TITLE
docker build speedups & memory reduction

### PR DESCRIPTION
**Pull request by casperdcl**
- docker: lots of optimisations + touch-ups
  + reorder commands to use more cached layers (quicker builds, quicker downloads, reduced disk space)
- python lint config: use the same line length everywhere
- ci: skip unnecessary PR runs

### CODE REVIEW

File: a2t-whisper/docker/cpu/Dockerfile [hunk 1]
```diff
@@ -1,31 +1,17 @@
 FROM python:3.10-slim-bullseye
-
-ARG MODEL_ID
-
 WORKDIR /usr/src/app/
 
-ENV PYTHONDONTWRITEBYTECODE 1
-ENV PYTHONUNBUFFERED 1
```
Comment (casperdcl) [inline a2t-whisper/docker/cpu/Dockerfile:8]: was this actually required? seen it in a few places from the root commit on this repo @filopedraz

File: cht-dolly-v2/docker/gpu/Dockerfile [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
+ARG MODEL_ID
 COPY download.py .
-
-RUN python3 download.py --model $MODEL_ID
+RUN python download.py --model $MODEL_ID
 
 COPY . .
-
 ENV MODEL_ID=$MODEL_ID
-ENV DEVICE=auto
```
Comment (casperdcl) [inline cht-dolly-v2/docker/gpu/Dockerfile:20]: is this required? different Dockerfiles use one of:

- `cpu`
- `cuda`
- `auto`
- nothing

File: cht-llama-cpp/docker/cpu/Dockerfile [hunk 3]
```diff
@@ -1,22 +1,17 @@
 FROM python:3.10-slim-bullseye
-
-ARG MODEL_ID
-
-RUN apt update && apt install -y libopenblas-dev ninja-build build-essential wget
-RUN python -m pip install --upgrade pip pytest cmake scikit-build setuptools
```
Comment (casperdcl) [inline cht-llama-cpp/docker/cpu/Dockerfile:6]: is this required? seen it in a few places

File: a2t-whisper/docker/cpu/Dockerfile [hunk 4]
```diff
@@ -1,31 +1,17 @@
...
-    git \
-    wget \
-    libatlas-base-dev \
-    ffmpeg
-
-COPY requirements.txt /usr/src/app/
-
-RUN pip3 install --no-cache-dir -r requirements.txt
+RUN apt update -qq && apt install -yqq --no-install-recommends \
+    build-essential git wget libatlas-base-dev ffmpeg \
```
Comment (casperdcl) [inline a2t-whisper/docker/cpu/Dockerfile:5]: can someone double-check all the `apt` installs in a ll the `Dockerfile`s? I suspect some of these deps aren't actually required (or maybe some are missing) - e.g. `libatlas-base-dev`?

File: cht-dolly-v2/docker/gpu/Dockerfile (hunk 2 above)
Comment (biswaroop1547) [reply cht-dolly-v2/docker/gpu/Dockerfile:20]: So when explicitly `cpu` or `cuda` is mentioned only these two cases should be required to make the model load on specified device on first try, I think `auto` cases can be removed.

Comment (filopedraz) [reply cht-dolly-v2/docker/gpu/Dockerfile:20]: Yep

File: cht-llama-cpp/docker/cpu/Dockerfile (hunk 3 above)
Comment (filopedraz) [reply cht-llama-cpp/docker/cpu/Dockerfile:6]: Maybe. But we will deprecate Vicuna and GPT4All, so no big deal.

File: a2t-whisper/docker/cpu/Dockerfile (hunk 4 above)
Comment (filopedraz) [reply a2t-whisper/docker/cpu/Dockerfile:6]: Let's create an issue for that :)

**Comment by filopedraz** [comment]
LGTM, but I would update the versions in the corresponding `build.sh` files or not @casperdcl @biswaroop1547 ?

**Comment by casperdcl** [comment]
Yes will probably bump them :)

**Review by filopedraz — APPROVED** [review]

**Comment by filopedraz** [comment]
I think we can merge and starting from October we will plan accordingly in order to deprecate some models.

**Pull request merged**